src/auth/jwt.strategy.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable, UnauthorizedException } from '@nestjs/common';
+import { PassportStrategy } from '@nestjs/passport';
+import { ExtractJwt, Strategy } from 'passport-jwt';
+import { Env } from 'src/env/index';
+import { JwtPayload } from './jwt-interface';
+import { RedisService } from '@liaoliaots/nestjs-redis';
+
+@Injectable()
+export class JwtStrategy extends PassportStrategy(Strategy, 'jwt') {
+  constructor(
+    private readonly redisService: RedisService,
+  ) {
+    super({
+      secretOrKey: Env.JWT.AUTH_TOKEN,
+      jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
+    });
+  }
+
+  async validate(payload: JwtPayload) {
+    const { userId } = payload;
+    const client = this.redisService.getClient();
+    const jwtToken = await client.get(`jwt:${userId}`);
+    if (!jwtToken) {
+      throw new UnauthorizedException();
+    }
+    return payload;
+  }
```

Comment:
PullProofAI

- The `JwtStrategy` class is well-structured and handles JWT authentication effectively.
- Good use of dependency injection for `RedisService`.
- Ensure error handling is robust, especially when interacting with Redis.
- Consider adding logging for better traceability in case of errors.

Overall, the `JwtStrategy` implementation looks solid and follows best practices for JWT authentication. Great job!


<!-- This is an auto-generated comment by PullProofAI -->